.github/workflows/codebuild.yml

```diff
@@ -17,22 +20,44 @@ jobs:
       id-token: write
       contents: read
     env:
-      event_name: ${{ github.event_name }}
       source_pr: pr/${{ github.event.pull_request.number }}
       source_sha: ${{ github.sha }}
+      pr_author: ${{ github.event.pull_request.user.login }}
     steps:
       - uses: actions/checkout@v4
+
+      - name: Get permissions
+        id: get_permission
+        if: github.event_name == 'pull_request_target'
+        uses: octokit/request-action@v2.x
+        with:
+          route: GET /repos/{repo}/collaborators/{author}/permission
+          repo: ${{ github.repository }}
+          author: ${{ github.event.pull_request.user.login }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Get credentials
         uses: aws-actions/configure-aws-credentials@v4.2.1
         with:
           role-to-assume: arn:aws:iam::024603541914:role/GitHubOIDCRole
           role-session-name: ${{ github.run_id }}
           aws-region: us-west-2
-      - name: Start Codebuild
+
+      - name: Start Codebuild for SHA
+        # This version runs when PRs are added to the merge queue or merged to main
+        if: github.event_name != 'pull_request_target'
+        run: ./codebuild/bin/start_codebuild.sh $source_sha
+
+      - name: Start Codebuild for PR
+        # This version runs when PRs are created or updated
+        if: github.event_name == 'pull_request_target'
         run: |
-          if [[ "$event_name" == "pull_request_target" ]]; then
-            source=$source_pr
+          permission=$(jq -r '.permission' <<< '${{ steps.get_permission.outputs.data }}')
+          echo "$pr_author has permission '$permission'".
+          if [[ "$permission" == "admin" || "$permission" == "write" ]]; then
```

Comment:
Technically, there is also "maintain", which falls between write and admin "https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization" - but we can iterate on what's really needed here.

Reply:
If you're curious, you can see the full output of the API call for me in the "Start Codebuild for PR" step here: https://github.com/lrstewart/s2n/actions/runs/16287636037/job/45989797539